ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -117,15 +140,36 @@ const MessageSpec& MessageDefinitionCache::load_message_spec(
     throw std::invalid_argument("Invalid package resource name: " +
                                 definition_identifier.package_resource_name);
   }
-  std::string package = match[1];
-  std::string share_dir = ament_index_cpp::get_package_share_directory(package);
-  std::ifstream file{share_dir + "/msg/" + match[2].str() +
-                     extension_for_format(definition_identifier.format)};
+  const std::string package = match[1].str();
+  const std::string filename = match[2].str() + extension_for_format(definition_identifier.format);
+
+  // Get the package share directory, or throw a PackageNotFoundError
+  const std::string share_dir = ament_index_cpp::get_package_share_directory(package);
+
+  // Get the rosidl_interfaces index contents for this package
+  std::string index_contents;
+  if (!ament_index_cpp::get_resource("rosidl_interfaces", package, index_contents)) {
+    throw DefinitionNotFoundError(definition_identifier.package_resource_name);
+  }
+
+  // Find the first line that ends with the filename we're looking for
+  const auto lines = split_string(index_contents);
+  const auto it = std::find_if(lines.begin(), lines.end(), [&filename](const std::string& line) {
+    return line.size() >= filename.size() &&
+           line.compare(line.size() - filename.size(), filename.size(), filename) == 0;
+  });
```

Comment:
Does this work if a package contains both `AAAAFoo.msg` and `Foo.msg`, and we're searching for `Foo.msg`?

Reply:
No, it will fail in that case 😭